modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -166,39 +176,30 @@ func validateMemo(ctx sdk.Context, memo string, allowedPacketDataList []string)
 	}
 
 	// if allowedPacketDataList has only 1 element and it equals AllowAllPacketDataKeys
-	// then accept all the packet data keys
+	// then accept all the memo strings
 	if len(allowedPacketDataList) == 1 && allowedPacketDataList[0] == AllowAllPacketDataKeys {
 		return nil
 	}
 
-	jsonObject := make(map[string]interface{})
-	err := json.Unmarshal([]byte(memo), &jsonObject)
-	if err != nil {
-		return err
-	}
-
 	gasCostPerIteration := ctx.KVGasConfig().IterNextCostFlat
 
-	for _, key := range allowedPacketDataList {
+	for _, allowedMemo := range allowedPacketDataList {
 		ctx.GasMeter().ConsumeGas(gasCostPerIteration, "transfer authorization")
 
-		_, ok := jsonObject[key]
-		if ok {
-			delete(jsonObject, key)
-		}
-	}
+		dst := &bytes.Buffer{}
+		json.Compact(dst, []byte(allowedMemo))
+		compactAllowedMemo := dst.String()
 
-	keys := make([]string, 0, len(jsonObject))
-	for k := range jsonObject {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
+		dst = &bytes.Buffer{}
+		json.Compact(dst, []byte(memo))
+		compactMemo := dst.String()
```

Comment:
Maybe this is over engineering it, but this compacts the JSON strings to prevent a mismatch due to empty spaces.